From b2d86d6ebeea2944e39fc73b327e0344b32771c8 Mon Sep 17 00:00:00 2001
From: Grzegorz Jaszczyk <jaz@semihalf.com>
Date: Mon, 26 Nov 2018 12:35:42 +0100
Subject: [PATCH 0916/1200] i2c: mv_i2c: warn if not supported speed mode
 chosen

Hitherto user could pass any speed value, while only two speeds mode were
currently supported. Warn if not supported speed mode is chosen.

Change-Id: Ia403382d479211b91169798da7a3b09a85d518dd
Signed-off-by: Grzegorz Jaszczyk <jaz@semihalf.com>
Reviewed-on: http://vgitil04.il.marvell.com:8080/61318
Tested-by: iSoC Platform CI <ykjenk@marvell.com>
Reviewed-by: Marcin Wojtas <mw@semihalf.com>
Reviewed-by: Kostya Porotchkin <kostap@marvell.com>
---
 drivers/i2c/mv_i2c.c | 14 +++++++++++---
 1 file changed, 11 insertions(+), 3 deletions(-)

diff --git a/drivers/i2c/mv_i2c.c b/drivers/i2c/mv_i2c.c
index cb0efa14d0..e0efe5851b 100644
--- a/drivers/i2c/mv_i2c.c
+++ b/drivers/i2c/mv_i2c.c
@@ -568,10 +568,18 @@ static int mv_i2c_set_bus_speed(struct udevice *bus, unsigned int speed)
 	struct mv_i2c_priv *priv = dev_get_priv(bus);
 	u32 val;
 
-	if (speed > 100000)
-		val = ICR_FM;
-	else
+	switch (speed) {
+	case 100000:
 		val = ICR_SM;
+		break;
+	case 400000:
+		val = ICR_FM;
+		break;
+	default:
+		printf("Only 100k and 400k modes supported\n");
+		return -EOPNOTSUPP;
+	}
+
 	clrsetbits_le32(&priv->base->icr, ICR_MODE_MASK, val);
 
 	return 0;
-- 
2.22.0

